firestore/integration_test_internal/src/firestore_test.cc

```diff
@@ -1348,7 +1405,105 @@ TEST_F(FirestoreIntegrationTest, RestartFirestoreLeadsToNewInstance) {
   delete db1;
 }
 
-TEST_F(FirestoreIntegrationTest, CanStopListeningAfterTerminate) {
+TEST_F(FirestoreTest, RestartFirestoreLeadsToNewNamedDatabaseInstance) {
+  // TODO(Mila): Remove the emulator env check after prod supports multiDB.
+  if (!firestore::IsUsingFirestoreEmulator()) {
+    GTEST_SKIP();
+  }
+
+  App* app = App::GetInstance();
+  Firestore* db1 = TestFirestoreWithDatabaseId(app->name(), "test-db");
+
+  // Create a document that we can use for verification later.
+  DocumentReference doc1 = db1->Collection("abc").Document();
+  const std::string doc_path = doc1.path();
+  EXPECT_THAT(doc1.Set({{"foo", FieldValue::String("bar")}}), FutureSucceeds());
+
+  // Terminate `db1` so that it will be removed from the instance cache.
```

Comment:
The test helper doesn't use the FirestoreCache found in `firestore/src/common/firestore.cc`. This undermines part of what you are trying to test.